src/som/interpreter/actors/SPromise.java

```diff
@@ -94,6 +106,29 @@ public final boolean isValue() {
     return false;
   }
 
+  /**
+   * Do not use for things other than serializing Promises.
+   *
+   * @return the value this promise was resolved to
+   */
+  public final Object getValue() {
+    assert VmSettings.SNAPSHOTS_ENABLED;
+    assert resolutionState == Resolution.SUCCESSFUL || resolutionState == Resolution.ERRONEOUS;
+    assert value != null;
+    return value;
+  }
+
+  /**
+   * Do not use for things other than deserializing Promises.
+   * This is necessary for circular object graphs.
+   */
+  public final void setValue(final Object value) {
```

Comment:
```suggestion
  public final void setValueFromSnapshot(final Object value) {
```

Or something along those lines would be better here, too.